python/ray/autoscaler/_private/monitor.py

```diff
@@ -192,12 +271,16 @@ def update_load_metrics(self):
                                  and self.autoscaler.config["provider"].get(
                                      "use_node_id_as_ip", False))
             if use_node_id_as_ip:
-                ip = str(int(total_resources.get("NODE_ID_AS_RESOURCE", 0)))
+                ip = resource_message.node_id.hex()
```

Comment:
just double-checking -- is the value for key "NODE_ID_AS_RESOURCE" definitely the same as the ray-internal node-id?
"NODE_ID_AS_RESOURCE" is not a standard Ray annotation is it?
Does this look ok, @yuduber ?

Reply:
This is actually removing NODE_ID_AS_RESOURCE (it seems to be a secret feature; we shouldn't have those).

Reply:
Agree with not having secret features. A better approach might be to document it, at least with a comment.

This is meant to allow using a node provider's node id in place of ip and was introduced to enable autoscaler integration with Peloton.
https://github.com/ray-project/ray/pull/17312#issue-952013550

Reply:
Fixed to be backwards compatible.